src/bufr_interface.f90

```diff
@@ -309,7 +309,8 @@ subroutine get_isc_c(isc_ptr, isc_size) bind(C, name='get_isc_f')
   type(c_ptr), intent(inout) :: isc_ptr
   integer(c_int), intent(out) :: isc_size
 
-  allocate(isc_f, source=isc)
+  allocate(isc_f(size(isc)))
```

Comment:
For this and all of the similar allocations and array copies below (i.e. for link, itp, typ, tag, and jmpb as well), you may want to consider only allocating and copying what you need, which is just the first ntab entries of the jump/link table, rather than the entire allocated size of the table in Fortran, which is 96000 or possibly even more.  For most reading and writing applications, ntab will be much smaller than that maximum number, and don't underestimate the possible runtime drag from having to copy all of those members from one array to another at runtime.  In fact, I'm running into this sort of thing in my work to consolidate all of the _4, _8, _d builds down to one, and where I therefore often have to copy BUFR messages (or portions thereof) from arrays of one build type to another during the interface step.

So, in summary, what you could do is just allocate an array of size ntab and then only copy those first ntab members from the Fortran side.  The value ntab is already stored directly within the moda_tables module, so you wouldn't even need to load a separate module.

Reply:
Thanks for the feedback @jbathegit! I'm not as worried about that for my code as I only read this table data 1 time for each BUFR file. Still I agree it would be good to minimize the resources that we are using, so I'll give your suggestion a shot.

Reply:
Ok, I made the changes you suggested and tested everything.